src/info/langs/mod.rs

```diff
@@ -1,211 +1,36 @@
-        assert_eq!(loc_by_language[&Language::Markdown], 300);
-    }
-
-    #[test]
-    fn deeply_nested_total_loc() {
-        let mut bash_code_stats = tokei::CodeStats::new();
-        // NOTE: When inside Markdown, comments should be counted as code
-        bash_code_stats.code = 5;
-        bash_code_stats.blanks = 1;
-        bash_code_stats.comments = 2;
-
-        let mut md_code_stats = tokei::CodeStats::new();
-        md_code_stats.code = 10;
-        md_code_stats.blanks = 2;
-        md_code_stats.comments = 4;
-        md_code_stats
-            .blobs
-            .insert(tokei::LanguageType::Bash, bash_code_stats);
-        // NOTE: This may break if tokei ever does more than just assign `name` to a field
-        let mut md_report = tokei::Report::new("/tmp/file.ipynb".into());
-        md_report.stats = md_code_stats;
-
-        let mut jupyter_notebook = tokei::Language::default();
-        jupyter_notebook
-            .children
-            .insert(tokei::LanguageType::Markdown, vec![md_report]);
-
-        let mut languages = tokei::Languages::new();
-        languages.insert(tokei::LanguageType::Jupyter, jupyter_notebook);
-
-        let loc_by_language = get_loc_by_language(&languages).unwrap();
-
-        assert_eq!(loc_by_language[&Language::Jupyter], 21);
-    }
-
-    // https://github.com/o2sh/onefetch/issues/966
-    #[test]
-    fn get_loc_by_language_should_not_panic_when_children_language_is_not_supported() {
-        let mut stylus_code_stats = tokei::CodeStats::new();
-        stylus_code_stats.code = 10;
-        stylus_code_stats.blanks = 2;
-        stylus_code_stats.comments = 4;
-
-        let mut stylus_report = tokei::Report::new("/tmp/file.vue".into());
-        stylus_report.stats = stylus_code_stats;
-
-        let mut vue = tokei::Language {
-            blanks: 50,
-            comments: 200,
-            code: 100,
-            ..Default::default()
-        };
-
-        vue.children
-            .insert(tokei::LanguageType::Stylus, vec![stylus_report]);
-
-        let mut languages = tokei::Languages::new();
-        languages.insert(tokei::LanguageType::Vue, vue);
-
-        let loc_by_language = get_loc_by_language(&languages).unwrap();
-
-        assert_eq!(loc_by_language[&Language::Vue], 110);
-    }
-
-    #[test]
-    fn test_get_loc_by_language_sorted() {
-        let mut map = HashMap::new();
-        map.insert(Language::Ada, 300);
-        map.insert(Language::Java, 40);
-        map.insert(Language::Rust, 1200);
-        map.insert(Language::Go, 8);
-
-        let sorted_map = sort_by_loc(map);
-
-        let expected_order = vec![
-            (Language::Rust, 1200),
-            (Language::Ada, 300),
-            (Language::Java, 40),
-            (Language::Go, 8),
-        ];
-        let actual_order: Vec<_> = sorted_map.into_iter().collect();
-
-        assert_eq!(expected_order, actual_order);
-    }
+    let analysis = match gengo.analyze("HEAD") {
```

Comment:
can't you use gango.analyze("HEAD")?;

Reply:
Same as https://github.com/o2sh/onefetch/pull/1152#discussion_r1313835411